main.py

```diff
@@ -226,6 +297,78 @@ def main():
+                    stateValue = states[stateId].stateValue
+                    stateToBeCalculated = states[stateId].id
+            checkStates.remove(stateToBeCalculated)
+
+            stateNeighbours = states[stateToBeCalculated].getNeighbours()
+            for stateNeighbour in stateNeighbours:
+                if stateNeighbour == GOAL:
+                    running = False
+                    print(str(states[stateToBeCalculated].distanceToStart + 1) + " moves needed!")
+                    previousState = GOAL
+                    currentState = states[stateToBeCalculated].state
+                    while stateToBeCalculated != 0:
+                        currentState = states[stateToBeCalculated].state
+                        displayedMove = ""
+                        for i in range(len(previousState)):
+                            if previousState[i] == 0 and currentState[i] > 0:
+                                displayedMove += str(COORDS[i])
+                                break
+                        displayedMove += " -> "
+                        for i in range(len(previousState)):
+                            if previousState[i] > 0 and currentState[i] == 0:
+                                displayedMove += str(COORDS[i])
+                                break
+                        print(displayedMove)
+                        previousState = currentState
+                        stateToBeCalculated = states[states[stateToBeCalculated].parent].id
+                    currentState = START
+                    displayedMove = ""
+                    for i in range(len(previousState)):
+                        if previousState[i] == 0 and currentState[i] > 0:
+                            displayedMove += str(COORDS[i])
+                            break
+                    displayedMove += " -> "
+                    for i in range(len(previousState)):
+                        if previousState[i] > 0 and currentState[i] == 0:
+                            displayedMove += str(COORDS[i])
+                            break
+                    print(displayedMove)
+                    break
+
+                stateInList = False
+                for state in states:
+                    if state.state == stateNeighbour:
+                        if state.distanceToStart > states[stateToBeCalculated].distanceToStart + 1:
+                            state.updateParent(stateToBeCalculated, states[stateToBeCalculated].distanceToStart + 1)
+                        stateInList = True
+                        break
+                if not stateInList:
+                    newId = len(states)
+                    states.append(State(newId, stateNeighbour, stateToBeCalculated, states[stateToBeCalculated].distanceToStart + 1))
+                    checkStates.append(newId)
+
+            if len(checkStates) == 0:
+                print("no solution found")
+                running = False
+
```

Comment:
[SUGGESTION] Outsource code and depending class to another file.
Therefore main file would not get overloaded and we have the chance to use directory structures to maybe hide bruteforce solutions on the root directory.
(General directory structure could be a pending issue for the repository)

Reply:
Maybe we should create an issue for that.

Reply:
Issue is the way